src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -544,16 +551,15 @@ public function preparePaymentsRequest(
                 //Getting line price
                 $price = $orderLine->getPrice();
 
-                //Getting order line information differently if it's a promotion or product
+                // Skip promotion line items.
                 if (empty($orderLine->getProductId()) && $orderLine->getType() === self::PROMOTION) {
-                    $productName = $orderLine->getDescription();
-                    $productNumber = $orderLine->getPayload()['promotionId'];
-                } else {
-                    $product = $this->getProduct($orderLine->getProductId(), $salesChannelContext->getContext());
-                    $productName = $product->getName();
-                    $productNumber = $product->getProductNumber();
+                    continue;
```

Comment:
So the plugin won't add the promotion to the invoice lines specifically, only if it is reflected as a line item with a negative amount. I see in this discussion that we were going to look into it so that we don't rely on the API to figure out the discount https://github.com/Adyen/adyen-shopware6/pull/153#discussion_r603838337

Reply:
Yes, we have a ticket to consult with Checkout on this